heartbeat/Filesystem

```diff
@@ -389,6 +389,82 @@ fstype_supported()
 	return $OCF_ERR_INSTALLED
 }
 
+#
+# Determine backend filesystem support state
+#
+backend_support()
+{
+	# network backended service type need retry timeout (default: 15)
+	# please note the timeout value here mean "15 times", not "15 secs"
+	local timeout
+	timeout=${OCF_RESKEY_CRM_meta_timeout:-15000}
+	timeout=$((timeout/1000))
+
+	# 0: success 1: failed (default)
+	local pass=1
+
+	while [ $timeout -gt 0 ]; do
+		case "$FSTYPE" in
+			glusterfs)
+				# it should failed if service not ready
+				# here we set a general timeout value for waiting response
+				# since gluster will have slow response when starting up
+				timeout 5 gluster volume status >/dev/null 2>&1 && pass=0
+				;;
+			ceph)
+				# it should failed if ceph not proper configure
+				# NOTE: degraded state is accepted
+				ceph health >/dev/null 2>&1 && pass=0
+				;;
+			overlay|overlayfs)
+				# it should fail immediately if kernel module not found
+				lsmod | grep -q "$FSTYPE" && pass=0
+				;;
+			nfs4|nfs|smbfs|cifs|gfs2|ocfs2)
+				# assume all these type were ready when start
+				pass=0
+				;;
+			*)
+				# $FSTYPE not support.
+				pass=1
```

Comment:
Perhaps set "pass" here to 0. The generic problem with this, not really coming up with this patch, is that with support for various filesystems spread through the code, we're bound to forget to update all places.